sled-agent/types/src/instance.rs

```diff
@@ -152,9 +184,6 @@ pub struct VmmUnregisterResponse {
 /// migration.
 #[derive(Copy, Clone, Debug, Deserialize, Serialize, JsonSchema)]
 pub struct InstanceMigrationTargetParams {
-    /// The Propolis ID of the migration source.
-    pub src_propolis_id: Uuid,
```

Comment:
It seems like a mistake to lose this, from the perspective of wishing we _had_ sent the expected UUID of the Downstairs we're connecting to from the Upstairs, and the bugs _not having it_ has caused.

Reply:
https://github.com/oxidecomputer/omicron/issues/6353 is a manifestation of the bug I'm talking about: an Upstairs constructed by a Pantry panicked when one of the regions it was connecting to did not match the others in the region set. The only information the Upstairs receives about Downstairs is SocketAddrs to connect to:

```
targets: [
  addr1, // downstairs A
  addr2, // downstairs B
  addr3  // downstairs C
]
```

When the Crucible agent creates and destroys Downstairs, it will aggressively reuse what ports are assigned to new Downstairs. If Nexus destroys Downstairs C, then creates a new Downstairs D (without doing anything else with the Agent), then Downstairs D will bind to the same port that Downstairs C did. If that happens then the targets array that the Upstairs was sent is instantly invalid, and constructing it will either 1) lead to a panic because the Downstairs D region is a different size than A and B, or 2) _not_ panic because Downstairs D is the same size as A and B, but it will be blank, leading to other issues.

Without any additional information in the targets array, the Upstairs has to panic: it would be invalid to continue and operate on regions with mismatched sizes. Adding the expected UUID would give the Upstairs the option to bail instead:

```
targets: [
  { uuidA, addr1 },
  { uuidB, addr2 },
  { uuidC, addr3 }
]
```

During the negotiation with the downstairs it would be able to compare the expected UUID in the targets array with what the Downstairs it's talking to is reporting, and fail to connect instead of panicking.